### DISCUSSION TITLE
How to check if compiler supports -include parameter?

### DISCUSSION THREAD

**Discussion by radioactiveman** [mesonbuild/meson#11409] [Q&A]
[Audacious](https://github.com/audacious-media-player) uses the following approach to include a generated header globally in every source file.

Is it possible to check if the compiler supports the `-include` parameter?
Even better would be a portable way to support other compilers as well.
The MSVC counter part for example is `/FI` if I understand their [docs](https://learn.microsoft.com/en-us/cpp/build/reference/fi-name-forced-include-file?view=msvc-170) correctly.

I have tried to use `cc.has_argument('-include')` but with no success.

```meson
cc = meson.get_compiler('c')
config_h = configure_file(input: 'config.h.meson', output: 'config.h', configuration: conf)
config_h_full_path = join_paths(meson.current_build_dir(), 'config.h')

if cc.get_id() in ['gcc', 'clang']
  add_project_arguments('-include', config_h_full_path, language: ['c', 'cpp'])
else
  error('Please implement -include handling for your chosen compiler.')
endif
```

**Comment by andy5995** [comment]
cxx = meson.get_compiler('cpp')
add_project_arguments(cxx.get_supported_arguments('-include'), language: 'cpp')

I don't think there's an error returned when the argument isn't supported; the arg just won't be added to the build.

**Comment by dcbaker** [comment]
Compiler.has_argument or has_multi_argument maybe?
